### DISCUSSION TITLE
`find-key` for tables/structs, `break` inside `eachp`?

### DISCUSSION THREAD

**Discussion by uvtc** [janet-lang/janet#900] [General]
I see that there's a `find-index` for arrays/tuples:

~~~janet
(def a [3 5 7 8 1])
(find-index even? a)  #=> 3
~~~

But no analogous `find-key` for maps. I tried this:

~~~janet
(defn find-key
  [pred tbl &opt dflt]
  (var res dflt) # result
  (eachp [k v] tbl
    (when (pred v)
      (set res k)
      (break)))  # <--- Hmm...
  res)
~~~

and it worked, but in the process I noticed that `break` works inside of `eachp`.

Is this supported behaviour, using `break` inside something other than a `while`?

**Comment by pepe** [comment]
`eachp` is a macro calling `each-template` https://github.com/janet-lang/janet/blob/a0ddfcb/src/boot/boot.janet#L432 a function generating the code implemented on top of `while`.
